### PR TITLE
RavenDB-6131 added TempPath and JournalsStoragePath to IndexingConfiguration

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

**Comment by ppekrol** [comment]
We should also implement http://issues.hibernatingrhinos.com/issue/RavenDB-6141

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -254,11 +265,20 @@ protected void Initialize(DocumentDatabase documentDatabase, IndexingConfigurati
...
+                    : null;
+
+                var journalPath = configuration.JournalsStoragePath != null
+                    ? Path.Combine(configuration.JournalsStoragePath, name)
+                    : null;
 
                 var options = configuration.RunInMemory
-                    ? StorageEnvironmentOptions.CreateMemoryOnly(indexPath)
+                    ? StorageEnvironmentOptions.CreateMemoryOnly(indexPath, indexTempPath)
                     : StorageEnvironmentOptions.ForPath(indexPath);
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Index.cs:282]: shouldn't we add it here in the `ForPath`?

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 2]
```diff
@@ -1289,7 +1309,7 @@ private IndexStats.MemoryStats GetMemoryStats()
         {
             var stats = new IndexStats.MemoryStats();
 
-            var indexPath = Path.Combine(Configuration.IndexStoragePath,
+            var indexPath = Path.Combine(Configuration.StoragePath,
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Index.cs:1312]: Need to also include the temp path folder as well

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by ppekrol** [comment]
I will fix those in a second, this was a mistake, git resolved conflict automatically in a wrong way.

**Pull request merged**